### ISSUE TITLE
Want host flash read over the network

### ISSUE THREAD

**Issue by rmustacc** [oxidecomputer/hubris#1957]
If you have humility access you can use the `qspi -r` or `qspi -R` options to read out data from the host qspi. This is probably using the ` HostFlash.read` operation and it would be nice if we could expose this via control-plane-agent so we could do the equivalents there.

**Comment by cbiffle** [comment]
I don't think there are any security implications to having unauthenticated read access to the QSPI, right? Its contents should be public bitstreams?

**Comment by jclulow** [comment]
> Its contents should be public bitstreams?

Yes, I believe that's correct.  It's a small piece of the output of Helios OS build process (open source), plus that zero page where we write the persistent BSU selection, as I recall?

**Comment by rmustacc** [comment]
> I don't think there are any security implications to having unauthenticated read access to the QSPI, right? Its contents should be public bitstreams?

I can't think of anything given the existence of public builds. The only thing that I think is worth thinking about is basically whether we want to phrase this as a read of the host image from a virtual slot that will skip over things like the persistent settings hubris uses or the memory-context restore or not. Part of the use of virtual slot is that I think (but am far from certain) for the initial use case I had in mind in an eSPI virtaulization world I'd want to read the virtualized thing.

Part of the reason for that is I'm not sure how much metadata or other information would end up in the SPI flash in the future that is derived information or something other than the OS image itself.

**Comment by mkeeter** [comment]
@labbott dupe of https://github.com/oxidecomputer/hubris/issues/2078 ?

**Comment by labbott** [comment]
> [@labbott](https://github.com/labbott) dupe of [#2078](https://github.com/oxidecomputer/hubris/issues/2078) ?

Not quite. #2078 was about wanting a Hash of contents. this is related to raw reading. Both are available now so I'm going to close the issue.
